build_runner/test/build_script_generate/builder_ordering_test.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import 'package:build_runner/src/build_script_generate/builder_ordering.dart';
+
+import '../common/build_configs.dart';
+
+void main() {
+  group('Builder ordering', () {
+    test('orders builders with `runs_before`', () async {
```

Comment:
nit: add some error validations tests for a cycle created by `runs_before` and `required_inputs`?

Reply:
nit: add a warning to the documentation that describes a the danger of creating a cycle.

Reply:
Added a separate section in the build_config README about builder ordering. Added a test that when there is a cycle it throws.